### ISSUE TITLE
Missing "catch" after "try" reports "finally expected", rather than noting the lack of a catch

### ISSUE THREAD

**Issue by jakebailey** [microsoft/TypeScript#47492]
This is totally silly, but I was writing something this weekend, brainfarted, and wrote the following code:

```ts
try {
    
} else {
    
}
```

[Playground Link](https://www.typescriptlang.org/play?ts=4.5.4#code/C4JwngBA3gUB8RgXwgUwDYGdXTg5QA)

It took me _way_ too long to figure out what I did wrong ("why is this squiggling?"), but the error that TS gave me was "'finally' expected", which didn't really push me in the right direction to see that what I actually forgot was to use the word "catch" (which is _way_ more common than `try {} finally {}`).

Maybe this is something that's a byproduct of that it's `try` then `catch` then `finally`, so `finally` is last, but I think I probably would have realized my error a little sooner if the message told me something different, like "catch expected" or "catch or finally expected".

**Comment by DanielRosenwasser** [comment]
If a `catch` wasn't already parsed, we should report

```
'catch' or 'finally' expected.
```

**Comment by mickey-stringer** [comment]
Created a PR for this [here](https://github.com/microsoft/TypeScript/pull/47504).
Without rewriting logic in `parser.ts`, the simplest approach seemed to be to take advantage of the alternate signature on `parseExpected` and pass it a custom `DiagnosticMessage`.

**Comment by DanielRosenwasser** [comment]
Thanks @mickey-stringer!
